website/blog/2023-11-05-workspace-folders.md

```diff
@@ -0,0 +1,63 @@
+
+All workspace folders are handled in metals separately and are oblivious of each other.
+E.g. for the following multi project structure
+
+```
+project1
+|- build.sbt
+|- src
+  |- Main.scala
+
+project2
+|- build.sbt
+|- src
+  |- Main.scala
+```
+
+we will keep two separate entities: one responsible for `project1`, and another one for `project2`.
+Upon receiving most requests metals will redirect them to the entity responsible for the project of interest. Of there are no other clues the project is chosen based on the currently opened file. E.g. if the user wants to insert an inferred type in the file `../project1/src/Main`, the request received by Metals will be redirected to the entity responsible for `project1`.
+
+For some requests we collect information from all the projects and send a joint result, e.g. when searching for workspace symbols.
+
+## How do I use the multi-root feature?
+
+In VSCode workspace folder support is achieved by [multi-root workspaces](https://code.visualstudio.com/docs/editor/multi-root-workspaces). To load multiple projects into a single workspace you can simply open one of the projects and add the other one using `File > Add Folder to Workspace` and then choosing the correct folder.
+
+![add-workspace-folder](https://i.imgur.com/LTYrx9V.gif)
+
+Now you have two projects loaded side by side, so you can easily see both and switch between them. All of the current metals functionality accommodates multiple projects, so you can use metals the same way as you did before. The biggest changes will be visible in the places where information from the whole workspace is collected, like workspace symbol search, test explorer, or metals doctor.
+
+![multi-root-tests](https://i.imgur.com/zWmmsC2.gif)
+
+The target project for a command is usually chosen based on the currently opened file. E.g. if you run `Switch build server` the command it will be executed for the project in focus. If no project is in focus the editor will explicitly ask for which project the command should be executed.
+
+![target-folder](https://i.imgur.com/tV7K822.gif)
+
+Finally, logs can still be found in the `.metals/metals.log` in the root of each project. Note, that for the time being all information is logged to all opened workspace folders, so anything logged for `project1` will also be visible in logs for `project2`.
+
+## Changes for the clients
+
+Since workspace folders are a part of the LSP for any client implementing this capability the multi-root support will work out of the box, however , there will be a few minor changes to accommodate the new approach.
```

Comment:
```suggestion
Since workspace folders are a part of the LSP for any client implementing this capability the multi-root support should work out of the box, however , there will be a few minor changes to accommodate the new approach.
```